### PR TITLE
KAFKA-7938: Fix test flakiness in DeleteConsumerGroupsTest

### PR DESCRIPTION
**Pull request by gwenshap**
This attempts to fix KAFKA-7938 and KAFKA-7946. 

I removed two tests:
- testDeleteWithShortInitialization basically didn't check the result and therefore always passed
- testDeleteCmdWithShortInitialization has no way to enforce that initialization is indeed short, and therefore sometimes FAILED because the group would be created before the CMD tried to delete it.

I thought the tests had limited value relative to the effort of figuring out a way to make the timing work.

I also fixed testDeleteCmdNonEmptyGroup and testDeleteNonEmptyGroup so they will validate that the group both exists and is non-empty before starting the test itself. I also added some extra information for future debugging sessions :)

I ran the tests LOTS of times to validate, but with flaky tests, it is hard to tell :)

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by gwenshap** [comment]
Forgot to mention:

1. The PR is against 2.2 because the goal is to unblock that release by fixing flaky tests. I can do a separate PR against trunk.

2. @vahidhashemian can you confirm the important (or lack thereof) of the tests I'm removing?

**Comment by gwenshap** [comment]
cc @mjsax

**Comment by gwenshap** [comment]
I heard you like flaky tests, so I put some flaky test failures in your flaky test fixes...

I think the failures are unrelated :)

File: core/src/test/scala/unit/kafka/admin/DeleteConsumerGroupsTest.scala [hunk 1]
```diff
@@ -88,10 +88,12 @@ class DeleteConsumerGroupsTest extends ConsumerGroupCommandTest {
     val service = getConsumerGroupService(cgcArgs)
 
     TestUtils.waitUntilTrue(() => {
-      service.listGroups().contains(group)
+      service.collectGroupMembers(false)._2.get.size == 1
     }, "The group did not initialize as expected.", maxRetries = 3)
 
     val result = service.deleteGroups()
+    println(result)
```
Comment (vahidhashemian) [inline core/src/test/scala/unit/kafka/admin/DeleteConsumerGroupsTest.scala:95]: Unintentional line here?

File: core/src/test/scala/unit/kafka/admin/DeleteConsumerGroupsTest.scala [hunk 2]
```diff
@@ -88,10 +88,12 @@ class DeleteConsumerGroupsTest extends ConsumerGroupCommandTest {
     val service = getConsumerGroupService(cgcArgs)
 
     TestUtils.waitUntilTrue(() => {
-      service.listGroups().contains(group)
+      service.collectGroupMembers(false)._2.get.size == 1
```
Comment (vahidhashemian) [inline core/src/test/scala/unit/kafka/admin/DeleteConsumerGroupsTest.scala:91]: Thanks for improving this check. Do you think it's worth doing the same in a few other unit tests in this file?

**Review by vahidhashemian — COMMENTED** [review]
Thanks @gwenshap for the PR. I'm fine with removing those flaky "short initialization" tests. They make more sense for describing groups anyway (where they were originally added).

I left a couple of minor comments inline.

File: core/src/test/scala/unit/kafka/admin/DeleteConsumerGroupsTest.scala (hunk 2 above)
Comment (gwenshap) [reply core/src/test/scala/unit/kafka/admin/DeleteConsumerGroupsTest.scala:91]: I went over the other tests, and none of them required a non-empty group specifically. Did I miss something?

File: core/src/test/scala/unit/kafka/admin/DeleteConsumerGroupsTest.scala (hunk 1 above)
Comment (gwenshap) [reply core/src/test/scala/unit/kafka/admin/DeleteConsumerGroupsTest.scala:95]: actually, quite intentional :) Those failures are difficult to reproduce, so more info in the console output of the test will help us next time.

I can add a comment if this is confusing.

Comment (vahidhashemian) [reply core/src/test/scala/unit/kafka/admin/DeleteConsumerGroupsTest.scala:95]: But this prints the output even when the test passes. If we want the output on failures only maybe that output can be added to the assertion error messages that follow. Unless there is another reason to have the output in either scenario?

File: core/src/test/scala/unit/kafka/admin/DeleteConsumerGroupsTest.scala (hunk 2 above)
Comment (vahidhashemian) [reply core/src/test/scala/unit/kafka/admin/DeleteConsumerGroupsTest.scala:91]: You're right, sorry my mistake.

File: core/src/test/scala/unit/kafka/admin/DeleteConsumerGroupsTest.scala (hunk 1 above)
Comment (gwenshap) [reply core/src/test/scala/unit/kafka/admin/DeleteConsumerGroupsTest.scala:95]: yeah, that makes sense. Will send the improvement tomorrow morning :)

**Comment by mjsax** [comment]
> The PR is against 2.2 because the goal is to unblock that release by fixing flaky tests. I can do a separate PR against trunk.

Why not to the PR against `trunk` and cherry-pick to `2.2` ?

**Comment by gwenshap** [comment]
> > The PR is against 2.2 because the goal is to unblock that release by fixing flaky tests. I can do a separate PR against trunk.
> 
> Why not to the PR against `trunk` and cherry-pick to `2.2` ?

will do.

**Comment by gwenshap** [comment]
Replaced by https://github.com/apache/kafka/pull/6312 which is against trunk.

**Pull request closed without merging**